src/drivers/src/led.c

```diff
@@ -64,20 +66,77 @@ static int led_polarity[] =
 };
 
 static bool isInit = 0;
-static bool enableBitmaskControl;
-static uint8_t ledBitmask;
+static uint8_t ledControlBitmask;
+static uint8_t ledLastState[LED_NUM];
+ledSwitch_t ledSwitchState;
 
-static void ledBitmaskCallback(void)
+static void ledRestoreSavedState(void)
 {
-  enableBitmaskControl = (ledBitmask & (1<<7)) != 0;
-
   for (int i = 0; i < LED_NUM; i++)
   {
-    ledSetOverride(i, ledBitmask & (1<<i));
+    ledSet(i, ledLastState[i]);
+  }
+}
+
+static void ledSetForce(led_t led, bool value)
+{
+  if (led>LED_NUM)
+    return;
+
+  if (led_polarity[led]==LED_POL_NEG)
+    value = !value;
+
+  if (led == LED_BLUE_NRF && isSyslinkUp())
+  {
+    SyslinkPacket slp;
+    slp.type = value ? SYSLINK_PM_LED_ON : SYSLINK_PM_LED_OFF;
+    slp.length = 0;
+    syslinkSendPacket(&slp);
+  }
+  else
+  {
+    if(value)
```

Comment:
nit: space after `if` and I prefer _måsvingar_ even on one-liners, but you decide.